samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/BotQnAMakerDialog.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.AI.QnA.Dialogs;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.BotBuilderSamples.Dialog
+{
+    /// <summary>
+    /// QnAMaker action builder class.  This class exists so that the base QnAMakerDialog can be supplied with
+    /// constructor arguments from configuration.
+    /// </summary>
+    public class BotQnAMakerDialog : QnAMakerDialog
+    {
+        public BotQnAMakerDialog(IConfiguration configuration)
+            :base(configuration["QnAKnowledgebaseId"], configuration["QnAEndpointKey"], BotQnAMakerDialog.GetHostname(configuration["QnAEndpointHostName"]))
+        {
+
+        }
+
+        private static string GetHostname(string hostname)
```

Comment:
Please use the URI Class to do this. The code here is error prone, and should not be used. :) 

It will fail, for example with "httpS". :)

Reply:
The intent of that method was to allow the user to specify either just a hostname or the full URL.  The Uri class does not do what is expected if someone doesn't specify a scheme.  Trying to introduce UriBuilder had similar gotchas.